pdns/ws-auth.cc

```diff
@@ -568,6 +568,10 @@ static void updateDomainSettingsFromDocument(UeberBackend& B, const DomainInfo&
   if (document["kind"].is_string()) {
     di.backend->setKind(zonename, DomainInfo::stringToKind(stringFromJson(document, "kind")));
   }
+  
+  if (arg()["default-soa-edit-api"] != "" && !document["soa_edit_api"].is_string()) {
```

Comment:
There is a side-effect here. `updateDomainSettingsFromDocument` is not only called when a domain is created, but also when a zone is modified with HTTP PUT. This means that existing zones with an empty (or non-existing) `SOA-EDIT-API` metadata will suddenly have this metadata data attached to them when editing e.g. records.

I think this default metadata should only be set on creation around [here](https://github.com/cyclops1982/pdns/blob/e6b32a057708bfd70f017ad6a264a03466b8ce47/pdns/ws-auth.cc#L1224-L1226)